install/first_google_boot.sh

```diff
@@ -249,26 +249,43 @@ function extract_spinnaker_gcr_credentials() {
   mkdir -p $(dirname $config_path)
   chown -R spinnaker:spinnaker $(dirname $config_path)
 
+  local gcr_enabled=$(get_instance_metadata_attribute "gcr_enabled")
   local gcr_account=$(get_instance_metadata_attribute "gcr_account")
-  if [ -n "$gcr_account" ]; then
-    # This downloads the JSON key for the specified service account
+
+  if [ -n "$gcr_enabled" ] || [ -n "$gcr_account" ]; then
+    echo "GCR enabled"
+
+    if [ -z "$gcr_account" ]; then
+      gcr_account=$(gcloud iam service-accounts list --filter="displayName:'Compute Engine default service account'" --format="value(email)")
+      clear_instance_metadata "gcr_account"
+    fi
+
+    echo "Extracting GCR credentials for email $gcr_account"
```

Comment:
That's fine - above if gcr_account isn't set it will be set to Compute Engine default service account, which exists as long as you have an instance running.